base/shared/src/main/scala/scalaz/typeclass/Show.scala

```diff
@@ -1,5 +1,22 @@
 package scalaz
+package typeclass
 
-abstract class Show[A] {
+/** A typeclass describing types which can be meaningfully represented as a `String`.
+  */
+trait Show[A] {
   def show(a: A): String
 }
+
+object Show extends ShowInstances with ShowSyntax {
+  def apply[A](implicit A: Show[A]): Show[A] = A
```

Comment:
`@inline` ?

Reply:
I feel like we really need a macro or compiler plugin like `@simulacrum.typeclass` to remove the boilerplate of all of this.

Reply:
Yeah. `simulacrum` doesn't do the zero-cost syntax; it could be made to support it or we could adapt it for our own purposes. That's not really my call, though; I'm just following what seems to be the current style.